web/src/components/PoiListItem.tsx

```diff
@@ -34,11 +35,13 @@ type PoiListItemProps = {
 
 const PoiListItem = ({ poi, distance, selectPoi }: PoiListItemProps): ReactElement => {
   const { t } = useTranslation('pois')
+  const [queryParams] = useSearchParams()
   const { title, category, slug } = poi
+  const slugWithQuery = `${slug}?${queryParams}`
 
   return (
     <ListItem disablePadding>
-      <StyledListItemButton onClick={selectPoi} id={slug} to={slug} component={Link} aria-label={title}>
+      <StyledListItemButton onClick={selectPoi} id={slug} to={slugWithQuery} component={Link} aria-label={title}>
```

Comment:
This could be solved actually in 2 ways:
1- The way I wrote it here.
2- Just removing the `to={slug} component={Link}` will solve that issue. I think the `component={Link}` is removing the the query and without it it will use onClick but we will lose the link accessibility thing.

Reply:
I think option 1. is the way to go here, its more declarative and probably better in terms of a11y. However, we now have duplicated logic (once in `to` and also in `selectPoi` (passed down from `Pois.tsx`)). I think we should delete `selectPoi` there in https://github.com/digitalfabrik/integreat-app/blob/main/web/src/components/Pois.tsx#L83. This will simplify code and reduce duplication. Then we can rename `selectPoi` here to just `onClick` as it will just adjust the scroll position.

Reply:
> I think option 1. is the way to go here, its more declarative and probably better in terms of a11y. However, we now have duplicated logic (once in `to` and also in `selectPoi` (passed down from `Pois.tsx`)). I think we should delete `selectPoi` there in https://github.com/digitalfabrik/integreat-app/blob/main/web/src/components/Pois.tsx#L83.

I can't delete `selectPoi` because switch poi depends on it for desktop 🤔